cmd/vic-machine/create/create.go

```diff
@@ -176,6 +198,166 @@ func (c *Create) Flags() []cli.Flag {
 	return flags
 }
 
+func zeroIPNet(n net.IPNet) bool {
+	return n.IP == nil && n.Mask == nil
+}
+
+func parseGatewaySpec(cg string) (string, *net.IPNet, error) {
+	if cg == "" {
+		return "", nil, cli.NewExitError("container network gateway is empty", 1)
+	}
+
+	mapped := strings.Split(cg, ":")
+	if len(mapped) != 2 {
+		return "", nil, cli.NewExitError(fmt.Sprintf("Container network gateway %s not in VSPHERE-NETWORK:GATEWAY format", cg), 1)
+	}
+
+	vnet := mapped[0]
+	if vnet == "" {
+		if vnet == "" {
+			return "", nil, cli.NewExitError(fmt.Sprintf("Container network not specified"), 1)
+		}
+	}
+
+	gw, subnet, err := net.ParseCIDR(mapped[1])
+	if err != nil {
+		return "", nil, cli.NewExitError(fmt.Sprintf("Invalid gateway %s specified for container network %s", mapped[1], vnet), 1)
+	}
+
+	return vnet, &net.IPNet{IP: gw, Mask: subnet.Mask}, nil
+}
+
+func (c *Create) processContainerNetworkGateways() (map[string]net.IPNet, error) {
+	gws := make(map[string]net.IPNet)
+	for _, cg := range c.containerNetworksGateway {
+		vnet, gw, err := parseGatewaySpec(cg)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, ok := gws[vnet]; ok {
```

Comment:
Is this actually a problem if we allow for IP ranges. Is it viable to have:
```
netA: 192.168.0.100-110
netB: 192.168.0.110-120
netC: 192.168.0.120-130
```

all with the gateway `192.168.0.1`?